mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/ModSeqProvider.java

```diff
@@ -43,6 +44,20 @@
      * @throws MailboxException
      */
     long nextModSeq(MailboxSession session, Mailbox mailbox) throws MailboxException;
+
+    /**
+     * Return the next mod-sequence which can be used for the {@link Mailbox}.
+     * Its important that the returned mod-sequence is higher then the last used and that the next call of this method does return a higher
+     * one. 
+     * 
+     * The first mod-seq must be >= 1
+     * 
+     * @param session
+     * @param mailboxId
+     * @return modSeq
+     * @throws MailboxException
+     */
+    long nextModSeq(MailboxSession session, MailboxId mailboxId) throws MailboxException;
```

Comment:
Because some implementations uses path lockers, that uses mailboxPath. Hence I can not simply rewrite it. @aduprat did the exact same thing.